src/pathfinding_igvc/test/path-finding-test.cpp

```diff
@@ -0,0 +1,219 @@
+#include <PathFinding.h>
+#include <gtest/gtest.h>
+
+TEST(PathFinding, testWeightedSum1) {
+    std::vector<float> testVec;
+    testVec.push_back(3);
+    testVec.push_back(3);
+    testVec.push_back(3);
+
+    EXPECT_EQ(5.5, PathFinding::weightedSum(testVec, 3));
+}
+
+TEST(PathFinding, testWeightedSum2) {
+    std::vector<float> testVec;
+    testVec.push_back(1);
+    testVec.push_back(2);
+    testVec.push_back(3);
+
+    EXPECT_EQ(1.0, PathFinding::weightedSum(testVec, 1));
+}
+
+// Case where robot starting orientation is forward and must travel a straight
+// path diagonal (along y=x line)
+TEST(PathFinding, testStraightPathToTwist) {
+    double xPos        = 0;
+    double yPos        = 0;
+    double orientation = 0;
+
+    nav_msgs::Path path_msg;
+    std::vector<geometry_msgs::PoseStamped> poses;
+    for (int i = 1; i < 10; i++) {
+        geometry_msgs::PoseStamped pose_stamped;
+        pose_stamped.pose.position.x = i;
+        pose_stamped.pose.position.y = i;
+        poses.push_back(pose_stamped);
+    }
+    path_msg.poses = poses;
+
+    geometry_msgs::Twist twist_msg =
+    PathFinding::pathToTwist(path_msg, xPos, yPos, orientation);
+    std::cout << "Straight Path" << std::endl;
+    std::cout << twist_msg.linear.x << std::endl;
+    std::cout << twist_msg.angular.z << std::endl;
+    std::cout << std::endl;
+    EXPECT_NEAR(0.875, twist_msg.linear.x, 0.01);
+    EXPECT_NEAR(0.785, twist_msg.angular.z, 0.01);
+}
+
+// Case where robot starting orientation is perpendicular to a straight path
+TEST(PathFinding, testPerpenPathToTwist) {
+    double xPos        = 0;
+    double yPos        = 0;
+    double orientation = M_PI / 2.0;
+
+    nav_msgs::Path path_msg;
+    std::vector<geometry_msgs::PoseStamped> poses;
+    for (int i = 1; i < 10; i++) {
+        geometry_msgs::PoseStamped pose_stamped;
+        pose_stamped.pose.position.x = i;
+        pose_stamped.pose.position.y = 0;
+        poses.push_back(pose_stamped);
+    }
+    path_msg.poses = poses;
+
+    geometry_msgs::Twist twist_msg =
+    PathFinding::pathToTwist(path_msg, xPos, yPos, orientation);
+    std::cout << "Perpen Path" << std::endl;
+    std::cout << twist_msg.linear.x << std::endl;
+    std::cout << twist_msg.angular.z << std::endl;
+    std::cout << std::endl;
```

Comment:
In general, we shouldn't really be printing anything in tests unless it directly relates to the test passing/failing (most of which gtest already tells us). This is to ensure that the output from full test runs (when we run all the tests), remains as compact and readable as possible.

Reply:
Fixed